Update README.md (#19)
brave/go-translate: README.md

# Translation server for brave

`go-translate` implements a translation server for use in brave-core written in Go.
It works on top of brave-hosted [Lingvanex On-premise Translation Server](https://lingvanex.com/translationserver/). It gets the requests from browsers in Chromium format,

> rewrite them to Lingvanex format, process requests and returns the result back to Chromium format.

Reviewer: rewrites, processes the requests
back in Chromium format

Author: ```suggestion
rewrites them to Lingvanex format, processes the requests and returns the result back in Chromium format.
```

## Build and run go-translate:

`make build`

## Local debugging

- Generate TLS credentials:
`openssl genrsa -out server.key 2048`
`openssl ecparam -genkey -name secp384r1 -out server.key`
`openssl req -new -x509 -sha256 -key server.key -out server.crt -days 3650`

> - Replace `err := srv.ListenAndServe()` to `err := srv.ListenAndServeTLS("server.crt", "server.key")` in `server/server.go`;

Reviewer: Replace with

Author: ```suggestion
- Replace `err := srv.ListenAndServe()` with `err := srv.ListenAndServeTLS("server.crt", "server.key")` in `server/server.go`;
```

- Set LNX_HOST (company VPN should be enabled):
  `export LNX_HOST=http://translate-lnx-dev-a4b82554457afe1c.elb.us-west-2.amazonaws.com:8080/api`

- Launch the local server: `make build`;

- Launch the browser with switches:
`--ignore-certificate-errors --translate-security-origin=https://127.0.0.1:8195/ --translate-script-url=https://127.0.0.1:8195/static/v1/element.js`.

- Other switches can be added if necessary (for example `--enable-features=UseBraveTranslateGo:update-languages/true` );

- Disable Shield on the tested sites (or globally), it cuts requests to localhost. `405 OPTIONS` errors also can be ignored;

> - If you have troubles check you can reach https://127.0.0.1:8195/static/v1/element.js in the tested browser.

Reviewer: check if you can

Author: ```suggestion
- If you have troubles check if you can reach https://127.0.0.1:8195/static/v1/element.js in the tested browser.
```
